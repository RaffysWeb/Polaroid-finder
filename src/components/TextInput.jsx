import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const TextInput = ({ placeholder, value, onChange, name, forwardedRef }) => (
  <StyledInput
    type="text"
    value={value}
    onChange={onChange}
    name={name}
    placeholder={placeholder}
    ref={forwardedRef}
  />
);

const StyledInput = styled.input`
  border-radius: 0.5rem;
  padding: 0.4rem 1rem;
  max-width: 25rem;
  border: 0.05rem solid lightblue;
  box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.23);

  &:focus {
    outline: none;
    border: 0.05rem solid blue;
  }
`;

TextInput.defaultProps = {
  value: '',
  placeholder: '',
  forwardedRef: null
};

TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string,
  placeholder: PropTypes.string,
  forwardedRef: PropTypes.instanceOf(Object),
  onChange: PropTypes.func.isRequired
};

export default TextInput;
