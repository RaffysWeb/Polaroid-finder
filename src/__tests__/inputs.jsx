import React,  from 'react';
import { create } from 'react-test-renderer';
import TextInput from '../components/TextInput';

describe('Inputs', () => {
  const testFunc = () => {
    return true;
  };

  test('TextInput it matches the snapshot', () => {
    const component = create(<TextInput name="test" onChange={testFunc} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
