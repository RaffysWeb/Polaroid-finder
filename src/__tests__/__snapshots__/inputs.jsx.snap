// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs TextInput it matches the snapshot 1`] = `
<input
  className="sc-bdVaJa kTNRlN"
  name="test"
  onChange={[Function]}
  placeholder=""
  type="text"
  value=""
/>
`;
